src/lib/runtime/ns/uts/uts.c

```diff
@@ -49,6 +60,10 @@ int _singularity_runtime_ns_uts(void) {
         singularity_message(ERROR, "Could not virtualize UTS namespace: %s\n", strerror(errno));
         ABORT(255);
     }
+    if ( hostname && sethostname(hostname, hostname_len) < 0 ) {
```

Comment:
Ahh, I see you are doing it in both ways. I think we should prefer to do it only via systemcall.